### ISSUE TITLE
Salesforce BigQuery Transfer | Parameter Issue

### ISSUE THREAD

**Issue by RomeoAva** [hashicorp/terraform-provider-google#18506]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.5.7
on MacOS
+ provider registry.terraform.io/hashicorp/google v5.34.0

### Affected Resource(s)

google_bigquery_data_transfer_config

### Terraform Configuration

```tf
terraform {
  backend "gcs" {
    bucket = ""
    prefix = "terraform/state"
  }

  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "5.34.0"
    }
    random = {
      source = "hashicorp/random"
    }
  }
}

```

### Expected Behavior

Hello, I am trying to create a [Salesforce Transfer](https://cloud.google.com/bigquery/docs/salesforce-transfer) for BQ using Terraform. Here is my current config to create that ressource:

```
resource "google_bigquery_data_transfer_config" "query_config" {
  display_name           = "import_salesforce"
  location               = "US"
  data_source_id         = "salesforce"
  destination_dataset_id = "analysis"
  params = {
    "connector.authentication.oauth.clientId"     = ""
    "connector.authentication.oauth.clientSecret" = ""
    "connector.authentication.username"           = ""
    "connector.authentication.password"           = ""
    "connector.authentication.securityToken"      = ""
    "assets"                                      = ["Project_ABC"]
  }
}
```

### Actual Behavior

When specifying a list in the `assets` parameter, terraform tells me:

```
 Inappropriate value for attribute "params": element "assets": string required.
```

When I try to convert it to a String, the transfer is created but on BigQuery I get the following error:

```
Incorrect assets set in transfer config : string_value: "Project_ABC" , assets should be a non-empty list of strings.
```

I also tried to json_encode the list, but same result on BQ side:

```
Incorrect assets set in transfer config : string_value: "[\"Project_ABC\"]" , assets should be a non-empty list of strings.
```

### Steps to reproduce

1. `terraform apply`

b/348691952

**Comment by ggtisc** [comment]
Confirmed issue!

According to [Google Cloud documentation](https://cloud.google.com/bigquery/docs/salesforce-transfer#bq) in the `bq section` describes the next format for the `assets` param: 

params='{"assets":["Account"]

But after running a `terraform apply` it returns the next error: 

`Inappropriate value for attribute "params": element "assets": string required.`

**Comment by wj-chen** [comment]
Wrapping just the assets list in `jsonencode` can get through the type validation. Can you try and see?
```
"assets" = jsonencode(["Project_ABC"])
```

**Comment by RomeoAva** [comment]
> Wrapping just the assets list in `jsonencode` can get through the type validation. Can you try and see?
> 
> ```
> "assets" = jsonencode(["Project_ABC"])
> ```

Hi, like mentioned in the last point of my « Actual Behavior » section, I tried that and it doesn’t work.

**Comment by wj-chen** [comment]
Thanks for the clarification. I see what you meant now - the transfer config could be created through Terraform but the actual transfer would fail for that error you mentioned. We'll continue to look into it.

**Comment by wj-chen** [comment]
PR has been merged and should be released to the Google provider as part of v5.40.0. A working config would be:
```
resource "google_bigquery_data_transfer_config" "salesforce_config" {
  display_name                 = "import_salesforce"
  location                           = "US"
  data_source_id               = "salesforce"
  destination_dataset_id = "analysis"
  params = {
    "connector.authentication.oauth.clientId"         = ""
    "connector.authentication.oauth.clientSecret" = ""
    "connector.authentication.username"                = ""
    "connector.authentication.password"                 = ""
    "connector.authentication.securityToken"         = ""
    "assets"                                                                     = "[Project_ABC, Project_XYZ]"
  }
}
```

**Comment by RomeoAva** [comment]
> PR has been merged and should be released to the Google provider as part of v5.40.0. A working config would be:
> 
> ```
> resource "google_bigquery_data_transfer_config" "salesforce_config" {
>   display_name           = "import_salesforce"
>   location               = "US"
>   data_source_id         = "salesforce"
>   destination_dataset_id = "analysis"
>   params = {
>     "connector.authentication.oauth.clientId"     = ""
>     "connector.authentication.oauth.clientSecret" = ""
>     "connector.authentication.username"           = ""
>     "connector.authentication.password"           = ""
>     "connector.authentication.securityToken"      = ""
>     "assets"                                      = "[Project_ABC, Project_XYZ]"
>   }
> }
> ```

Thanks for the work and help !
